app/Http/Respositories/Eloquent/InvoiceRepository.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace App\Http\Respositories\Eloquent;
+
+use App\Http\Respositories\InvoiceRepositoryInterface;
+use App\Models\Invoice;
+
+class InvoiceRepository extends BaseRepository implements InvoiceRepositoryInterface
+{
+    protected $model;
+
+    /**
+     * @param $model
+     */
+    public function __construct(Invoice $model)
+    {
+        parent::__construct($model);
+        $this->model = $model;
+    }
+
+    public function deleteWithRelationship($request)
```

Comment:
get id from request in service, do not put the request in the repository
and what goal of this function?